bin/sozo/src/commands/clean.rs

```diff
@@ -41,18 +35,100 @@ impl CleanArgs {
         let profile_name =
             ws.current_profile().expect("Scarb profile is expected at this point.").to_string();
 
-        let clean_manifests_abis = self.manifests_abis || !self.artifacts;
-        let clean_artifacts = self.artifacts || !self.manifests_abis;
+        // Manifest path is always a file, we can unwrap safely to get the
+        // parent folder.
+        let manifest_dir = ws.manifest_path().parent().unwrap().to_path_buf();
 
-        if clean_manifests_abis {
-            let manifest_dir = ws.manifest_path().parent().unwrap().to_path_buf();
-            self.clean_manifests_abis(&manifest_dir, &profile_name)?;
-        }
+        let profile_dir = manifest_dir.join(MANIFESTS_DIR).join(profile_name);
 
-        if clean_artifacts {
-            scarb::ops::clean(config)?;
+        // By default, this command cleans the build manifests and scarb artifacts.
+        scarb::ops::clean(config)?;
+        self.clean_manifests(&profile_dir)?;
+
+        if self.all && profile_dir.exists() {
+            fs::remove_dir_all(profile_dir)?;
         }
 
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use dojo_test_utils::compiler;
+    use dojo_test_utils::sequencer::{get_default_test_starknet_config, TestSequencer};
+    use sozo_ops::migration;
+
+    use super::*;
+
+    #[test]
+    fn test_clean() {
+        let source_project = "../../examples/spawn-and-move/Scarb.toml";
+
+        // Build a completely new project in it's own directory.
+        let (temp_project_dir, config, _) = compiler::copy_build_project_temp(source_project, true);
+
+        let sequencer = config.tokio_handle().block_on(async {
+            TestSequencer::start(Default::default(), get_default_test_starknet_config()).await
+        });
```

Comment:
maybe we can start using the katana [runner](https://github.com/dojoengine/dojo/tree/main/crates/katana/runner) for tests that need to instantiate katana ?

Reply:
100% we should work to do this everywhere. it will allow us to decouple katana versioning further

Reply:
> maybe we can start using the katana [runner](https://github.com/dojoengine/dojo/tree/main/crates/katana/runner) for tests that need to instantiate katana ? 

Good point, will do it here to have an example for others and following revamp testing by using in tests. 👍

Reply:
Modified in e02a839.